admin_tests/base/test_utils.py

```diff
@@ -158,3 +165,48 @@ def test_change_embargo_date(self):
         assert_almost_equal(self.registration.embargo.end_date, self.date_valid2, delta=datetime.timedelta(days=1))
 
         # Add a test to check privatizing
+
+site = AdminSite()
+
+class TestGroupCollectionsPreprints(AdminTestCase):
+
+    @pytest.mark.enable_bookmark_creation
+    def test_admin_app_groups(self):
+        user = UserFactory()
+        user.save()
+        request = RequestFactory()
+        get_request = request.get('/admin/osf/osfuser/{}/change'.format(user.id))
+        get_request.user = user
+        post_request = request.post('/admin/osf/osfuser/{}/change'.format(user.id))
+        post_request.user = user
+        preprint = PreprintFactory(creator=user)
+        preprint.save()
+        admin = OSFUserAdmin(OSFUser, site)
+        formfield = (admin.formfield_for_manytomany(OSFUser.groups.field, request=get_request))
+        queryset = formfield.queryset
+
+        collections_group = Group.objects.filter(name__startswith='collections_')[0]
+        assert(collections_group not in queryset)
+
+        preprint_group = Group.objects.filter(name__startswith='preprint_')[0]
+        assert(preprint_group not in queryset)
+
+        form = admin.get_form(request=post_request, obj=user)
+        data_dict = model_to_dict(user)
+        post_form = form(data_dict, instance=user)
+
+        #Removing any fields which are causing JSON related errors
+        for field in post_form.errors.keys():
+            if field == 'groups':
+                data_dict['groups'] = []
+            else:
+                data_dict[field] = '{}'
+        post_form = form(data_dict, instance=user)
+        assert(post_form.is_valid())
+        post_form.save(commit=False)
+        qdict = QueryDict('', mutable=True)
+        qdict.update(data_dict)
+        post_request.POST = qdict
+        admin.save_related(request=post_request, form=post_form, formsets=[], change=True)
+        assert(len(list(user.groups.filter(name__startswith='collections_'))) > 0)
+        assert(len(list(user.groups.filter(name__startswith='preprint_'))) > 0)
```

Comment:
Can you assert that both the preprint_group and the collection_group are in user_groups instead of that the length is bigger than 0?